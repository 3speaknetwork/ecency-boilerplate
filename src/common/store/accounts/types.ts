import { ExtendedAccount } from "@esteemapp/dhive";

export interface Account extends ExtendedAccount {}

export interface State {
  data: Account;
  loading: boolean;
  error: boolean;
}

export enum ActionTypes {
  FETCH = "@account/FETCH",
  FETCHED = "@account/FETCHED",
  FETCH_ERROR = "@account/FETCH_ERROR"
}

export interface FetchAction {
  type: ActionTypes.FETCH;
}

export interface FetchedAction {
  type: ActionTypes.FETCHED;
  data: Account;
}

export interface FetchErrorAction {
  type: ActionTypes.FETCH_ERROR;
}

export type Actions = FetchAction | FetchedAction | FetchErrorAction;