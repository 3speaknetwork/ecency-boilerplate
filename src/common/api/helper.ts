import isElectron from "../util/is-electron";

import defaults from "../constants/defaults.json";

export const apiBase = (endpoint: string): string => {
    let base = '';

    if (isElectron()) {
        if (process.env.NODE_ENV === "development") {
            base = "http://localhost:3000";
        } else {
            base = defaults.base;
        }
    }

    return `${base}${endpoint}`;
}
