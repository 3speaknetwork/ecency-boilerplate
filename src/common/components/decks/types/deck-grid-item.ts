interface BaseDeckGridItem {
  key: number;
  type: "u" | "ac" | "co" | "w" | "n" | "tr" | "to" | "s" | "cu";
  settings: unknown;
}

export interface UserDeckGridItem extends BaseDeckGridItem {
  type: "u";
  settings: {
    username: string;
    contentType: string;
  };
}

export type DeckGridItem = BaseDeckGridItem | UserDeckGridItem;
