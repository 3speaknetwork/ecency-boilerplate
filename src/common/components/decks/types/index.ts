export * from "./deck-grid-item";
export * from "./deck-grid";
