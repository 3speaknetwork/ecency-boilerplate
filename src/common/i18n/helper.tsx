import React from "react";

import {_t} from "./index";

interface Props {
    k: string,
    args: {},
    children: JSX.Element
}

export class Tsx extends React.Component<Props> {
    render() {
        const {k, args} = this.props;
        const {children} = this.props;
        return React.cloneElement(children, {dangerouslySetInnerHTML: {__html: _t(k, args)}});
    }
}
