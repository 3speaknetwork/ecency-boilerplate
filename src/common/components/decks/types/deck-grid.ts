import { DeckGridItem } from "./deck-grid-item";

export interface DeckGrid {
  columns: DeckGridItem[];
}
