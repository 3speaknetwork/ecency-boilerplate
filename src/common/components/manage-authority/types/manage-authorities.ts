import { Account } from "../../../store/accounts/types";

export const Keytype = {
  Owner: "owner",
  Active: "active",
  Posting: "posting"
};

export interface PublicKeys {
  publicOwnerKey?: any;
  publicActiveKey?: any;
  publicPostingKey?: any;
}

export interface AccountDataType {
  postingsAuthority: Array<any>;
  posting: Array<any>;
  owner: Array<any>;
  active: Array<any>;
  weight: number;
  memokey: string;
  account: Account;
  PublicKeys: {
    publicOwnerKey: any;
    publicActiveKey: any;
    publicPostingKey: any;
  };
}

export const actionType = {
  Revoke: "revoke"
};
