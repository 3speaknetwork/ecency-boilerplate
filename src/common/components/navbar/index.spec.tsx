import React from "react";
import {StaticRouter} from "react-router-dom";
import TestRenderer from "react-test-renderer";
import {createBrowserHistory, createLocation} from "history";

import NavBar from "./index";

import {Theme} from "../../store/global/types";

import {globalInstance, TrendingTagsInstance, UiInstance, notificationsInstance1, activeUserInstance} from "../../helper/test-helper";


const defProps = {
    history: createBrowserHistory(),
    location: createLocation({}),
    global: globalInstance,
    trendingTags: TrendingTagsInstance,
    users: [],
    activeUser: null,
    ui: UiInstance,
    notifications: notificationsInstance1,
    fetchTrendingTags: () => {
    },
    toggleTheme: () => {
    },
    addUser: () => {
    },
    setActiveUser: () => {
    },
    updateActiveUser: () => {
    },
    addAccount: () => {
    },
    deleteUser: () => {
    },
    fetchNotifications: () => {
    },
    fetchUnreadNotificationCount: () => {
    },
    setNotificationsFilter: () => {
    },
    markNotifications: () => {
    },
    toggleUIProp: () => {
    },
    muteNotifications: () => {
    },
    unMuteNotifications: () => {
    },
};

it("(1) Default render", () => {
    const component = <NavBar {...defProps} />;

    const renderer = TestRenderer.create(
        <StaticRouter location="/" context={{}}>
            {component}
        </StaticRouter>
    );

    expect(renderer.toJSON()).toMatchSnapshot();
});


it("(2) Night Theme", () => {
    const props = {
        ...defProps,
        ...{
            global: {
                ...globalInstance,
                theme: Theme.night
            }
        }
    }
    const component = <NavBar {...props} />;

    const renderer = TestRenderer.create(
        <StaticRouter location="/" context={{}}>
            {component}
        </StaticRouter>
    );

    expect(renderer.toJSON()).toMatchSnapshot();
});


it("(3) With active user", () => {
    const props = {
        ...defProps,
        ...{
            activeUser: {...activeUserInstance}
        }
    }
    const component = <NavBar {...props} />;

    const renderer = TestRenderer.create(
        <StaticRouter location="/" context={{}}>
            {component}
        </StaticRouter>
    );

    expect(renderer.toJSON()).toMatchSnapshot();
});
