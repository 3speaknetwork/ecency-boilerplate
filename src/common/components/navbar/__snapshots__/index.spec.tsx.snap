// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(1) Default render 1`] = `
Array [
  <div
    className="nav-bar-toggle"
    onClick={[Function]}
  >
    <svg
      viewBox="0 0 24 24"
    >
      <path
        d="M3,6H21V8H3V6M3,11H21V13H3V11M3,16H21V18H3V16Z"
        fill="currentColor"
      />
    </svg>
  </div>,
  <div
    className="nav-bar-sm"
  >
    <div
      className="brand"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt="Logo"
          className="logo"
          src="logo-circle.svg"
        />
      </a>
    </div>
    <div
      className="text-menu"
    >
      <a
        className="menu-item"
        href="/discover"
        onClick={[Function]}
      >
        Discover
      </a>
      <a
        className="menu-item"
        href="/communities"
        onClick={[Function]}
      >
        Communities
      </a>
    </div>
  </div>,
  <div
    className="nav-bar"
  >
    <div
      className="nav-bar-inner"
    >
      <div
        className="brand"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <img
            alt="Logo"
            className="logo"
            src="logo-circle.svg"
          />
        </a>
      </div>
      <div
        className="text-menu"
      >
        <a
          className="menu-item"
          href="/discover"
          onClick={[Function]}
        >
          Discover
        </a>
        <a
          className="menu-item"
          href="/communities"
          onClick={[Function]}
        >
          Communities
        </a>
      </div>
      <div
        className="flex-spacer"
      />
      <div
        className="search-bar"
      >
        <div
          className="suggestion"
        >
          <div
            className="search-box"
          >
            <span
              className="prepend"
            >
              <svg
                viewBox="0 0 24 24"
              >
                <path
                  d="M9.5,3A6.5,6.5 0 0,1 16,9.5C16,11.11 15.41,12.59 14.44,13.73L14.71,14H15.5L20.5,19L19,20.5L14,15.5V14.71L13.73,14.44C12.59,15.41 11.11,16 9.5,16A6.5,6.5 0 0,1 3,9.5A6.5,6.5 0 0,1 9.5,3M9.5,5C7,5 5,7 5,9.5C5,12 7,14 9.5,14C12,14 14,12 14,9.5C14,7 12,5 9.5,5Z"
                  fill="currentColor"
                />
              </svg>
            </span>
            <input
              autoComplete="off"
              className="form-control"
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Search 10,000,000 posts"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        className="switch-menu"
      >
        <div
          className="switch-language"
        >
          <div
            className="custom-dropdown float-right"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="dropdown-btn"
            >
              
              <div
                className="menu-down"
              >
                <svg
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12.87,15.07L10.33,12.56L10.36,12.53C12.1,10.59 13.34,8.36 14.07,6H17V4H10V2H8V4H1V6H12.17C11.5,7.92 10.44,9.75 9,11.35C8.07,10.32 7.3,9.19 6.69,8H4.69C5.42,9.63 6.42,11.17 7.67,12.56L2.58,17.58L4,19L9,14L12.11,17.11L12.87,15.07M18.5,10H16.5L12,22H14L15.12,19H19.87L21,22H23L18.5,10M15.88,17L17.5,12.67L19.12,17H15.88Z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          className="switch-theme"
          onClick={[Function]}
          title="Night Theme"
        >
          <svg
            viewBox="0 0 24 24"
          >
            <path
              d="M12,18V6A6,6 0 0,1 18,12A6,6 0 0,1 12,18M20,15.31L23.31,12L20,8.69V4H15.31L12,0.69L8.69,4H4V8.69L0.69,12L4,15.31V20H8.69L12,23.31L15.31,20H20V15.31Z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
      <div
        className="btn-menu"
      >
        <div
          className="login-required"
        >
          <button
            className="btn-login btn btn-outline-primary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Login
          </button>
          <a
            className="btn btn-primary"
            href="/signup"
            onClick={[Function]}
          >
            Signup
          </a>
        </div>
        <div
          className="submit-post"
        >
          <a
            className="btn btn-outline-primary"
            href="/submit"
            onClick={[Function]}
            title="Create Post"
          >
            <svg
              viewBox="0 0 24 24"
            >
              <path
                d="M14.06,9L15,9.94L5.92,19H5V18.08L14.06,9M17.66,3C17.41,3 17.15,3.1 16.96,3.29L15.13,5.12L18.88,8.87L20.71,7.04C21.1,6.65 21.1,6 20.71,5.63L18.37,3.29C18.17,3.09 17.92,3 17.66,3M14.06,6.19L3,17.25V21H6.75L17.81,9.94L14.06,6.19Z"
                fill="currentColor"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
    <audio
      autoPlay={false}
      id="notification-audio"
      muted={true}
      src="notification.mp3"
      style={
        Object {
          "display": "none",
        }
      }
    />
  </div>,
]
`;

exports[`(2) Night Theme 1`] = `
Array [
  <div
    className="nav-bar-toggle"
    onClick={[Function]}
  >
    <svg
      viewBox="0 0 24 24"
    >
      <path
        d="M3,6H21V8H3V6M3,11H21V13H3V11M3,16H21V18H3V16Z"
        fill="currentColor"
      />
    </svg>
  </div>,
  <div
    className="nav-bar-sm"
  >
    <div
      className="brand"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt="Logo"
          className="logo"
          src="logo-circle.svg"
        />
      </a>
    </div>
    <div
      className="text-menu"
    >
      <a
        className="menu-item"
        href="/discover"
        onClick={[Function]}
      >
        Discover
      </a>
      <a
        className="menu-item"
        href="/communities"
        onClick={[Function]}
      >
        Communities
      </a>
    </div>
  </div>,
  <div
    className="nav-bar"
  >
    <div
      className="nav-bar-inner"
    >
      <div
        className="brand"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <img
            alt="Logo"
            className="logo"
            src="logo-circle.svg"
          />
        </a>
      </div>
      <div
        className="text-menu"
      >
        <a
          className="menu-item"
          href="/discover"
          onClick={[Function]}
        >
          Discover
        </a>
        <a
          className="menu-item"
          href="/communities"
          onClick={[Function]}
        >
          Communities
        </a>
      </div>
      <div
        className="flex-spacer"
      />
      <div
        className="search-bar"
      >
        <div
          className="suggestion"
        >
          <div
            className="search-box"
          >
            <span
              className="prepend"
            >
              <svg
                viewBox="0 0 24 24"
              >
                <path
                  d="M9.5,3A6.5,6.5 0 0,1 16,9.5C16,11.11 15.41,12.59 14.44,13.73L14.71,14H15.5L20.5,19L19,20.5L14,15.5V14.71L13.73,14.44C12.59,15.41 11.11,16 9.5,16A6.5,6.5 0 0,1 3,9.5A6.5,6.5 0 0,1 9.5,3M9.5,5C7,5 5,7 5,9.5C5,12 7,14 9.5,14C12,14 14,12 14,9.5C14,7 12,5 9.5,5Z"
                  fill="currentColor"
                />
              </svg>
            </span>
            <input
              autoComplete="off"
              className="form-control"
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Search 10,000,000 posts"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        className="switch-menu"
      >
        <div
          className="switch-language"
        >
          <div
            className="custom-dropdown float-right"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="dropdown-btn"
            >
              
              <div
                className="menu-down"
              >
                <svg
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12.87,15.07L10.33,12.56L10.36,12.53C12.1,10.59 13.34,8.36 14.07,6H17V4H10V2H8V4H1V6H12.17C11.5,7.92 10.44,9.75 9,11.35C8.07,10.32 7.3,9.19 6.69,8H4.69C5.42,9.63 6.42,11.17 7.67,12.56L2.58,17.58L4,19L9,14L12.11,17.11L12.87,15.07M18.5,10H16.5L12,22H14L15.12,19H19.87L21,22H23L18.5,10M15.88,17L17.5,12.67L19.12,17H15.88Z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          className="switch-theme"
          onClick={[Function]}
          title="Day Theme"
        >
          <svg
            viewBox="0 0 24 24"
          >
            <path
              d="M12,18V6A6,6 0 0,1 18,12A6,6 0 0,1 12,18M20,15.31L23.31,12L20,8.69V4H15.31L12,0.69L8.69,4H4V8.69L0.69,12L4,15.31V20H8.69L12,23.31L15.31,20H20V15.31Z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
      <div
        className="btn-menu"
      >
        <div
          className="login-required"
        >
          <button
            className="btn-login btn btn-outline-primary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Login
          </button>
          <a
            className="btn btn-primary"
            href="/signup"
            onClick={[Function]}
          >
            Signup
          </a>
        </div>
        <div
          className="submit-post"
        >
          <a
            className="btn btn-outline-primary"
            href="/submit"
            onClick={[Function]}
            title="Create Post"
          >
            <svg
              viewBox="0 0 24 24"
            >
              <path
                d="M14.06,9L15,9.94L5.92,19H5V18.08L14.06,9M17.66,3C17.41,3 17.15,3.1 16.96,3.29L15.13,5.12L18.88,8.87L20.71,7.04C21.1,6.65 21.1,6 20.71,5.63L18.37,3.29C18.17,3.09 17.92,3 17.66,3M14.06,6.19L3,17.25V21H6.75L17.81,9.94L14.06,6.19Z"
                fill="currentColor"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
    <audio
      autoPlay={false}
      id="notification-audio"
      muted={true}
      src="notification.mp3"
      style={
        Object {
          "display": "none",
        }
      }
    />
  </div>,
]
`;

exports[`(3) With active user 1`] = `
Array [
  <div
    className="nav-bar-toggle"
    onClick={[Function]}
  >
    <svg
      viewBox="0 0 24 24"
    >
      <path
        d="M3,6H21V8H3V6M3,11H21V13H3V11M3,16H21V18H3V16Z"
        fill="currentColor"
      />
    </svg>
  </div>,
  <div
    className="nav-bar-sm"
  >
    <div
      className="brand"
    >
      <a
        href="/@foo/feed"
        onClick={[Function]}
      >
        <img
          alt="Logo"
          className="logo"
          src="logo-circle.svg"
        />
      </a>
    </div>
    <div
      className="text-menu"
    >
      <a
        className="menu-item"
        href="/discover"
        onClick={[Function]}
      >
        Discover
      </a>
      <a
        className="menu-item"
        href="/communities"
        onClick={[Function]}
      >
        Communities
      </a>
    </div>
  </div>,
  <div
    className="nav-bar"
  >
    <div
      className="nav-bar-inner"
    >
      <div
        className="brand"
      >
        <a
          href="/@foo/feed"
          onClick={[Function]}
        >
          <img
            alt="Logo"
            className="logo"
            src="logo-circle.svg"
          />
        </a>
      </div>
      <div
        className="text-menu"
      >
        <a
          className="menu-item"
          href="/discover"
          onClick={[Function]}
        >
          Discover
        </a>
        <a
          className="menu-item"
          href="/communities"
          onClick={[Function]}
        >
          Communities
        </a>
      </div>
      <div
        className="flex-spacer"
      />
      <div
        className="search-bar"
      >
        <div
          className="suggestion"
        >
          <div
            className="search-box"
          >
            <span
              className="prepend"
            >
              <svg
                viewBox="0 0 24 24"
              >
                <path
                  d="M9.5,3A6.5,6.5 0 0,1 16,9.5C16,11.11 15.41,12.59 14.44,13.73L14.71,14H15.5L20.5,19L19,20.5L14,15.5V14.71L13.73,14.44C12.59,15.41 11.11,16 9.5,16A6.5,6.5 0 0,1 3,9.5A6.5,6.5 0 0,1 9.5,3M9.5,5C7,5 5,7 5,9.5C5,12 7,14 9.5,14C12,14 14,12 14,9.5C14,7 12,5 9.5,5Z"
                  fill="currentColor"
                />
              </svg>
            </span>
            <input
              autoComplete="off"
              className="form-control"
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Search 10,000,000 posts"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        className="switch-menu"
      >
        <div
          className="switch-language"
        >
          <div
            className="custom-dropdown float-right"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="dropdown-btn"
            >
              
              <div
                className="menu-down"
              >
                <svg
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12.87,15.07L10.33,12.56L10.36,12.53C12.1,10.59 13.34,8.36 14.07,6H17V4H10V2H8V4H1V6H12.17C11.5,7.92 10.44,9.75 9,11.35C8.07,10.32 7.3,9.19 6.69,8H4.69C5.42,9.63 6.42,11.17 7.67,12.56L2.58,17.58L4,19L9,14L12.11,17.11L12.87,15.07M18.5,10H16.5L12,22H14L15.12,19H19.87L21,22H23L18.5,10M15.88,17L17.5,12.67L19.12,17H15.88Z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          className="switch-theme"
          onClick={[Function]}
          title="Night Theme"
        >
          <svg
            viewBox="0 0 24 24"
          >
            <path
              d="M12,18V6A6,6 0 0,1 18,12A6,6 0 0,1 12,18M20,15.31L23.31,12L20,8.69V4H15.31L12,0.69L8.69,4H4V8.69L0.69,12L4,15.31V20H8.69L12,23.31L15.31,20H20V15.31Z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
      <div
        className="btn-menu"
      >
        <div
          className="submit-post"
        >
          <a
            className="btn btn-outline-primary"
            href="/submit"
            onClick={[Function]}
            title="Create Post"
          >
            <svg
              viewBox="0 0 24 24"
            >
              <path
                d="M14.06,9L15,9.94L5.92,19H5V18.08L14.06,9M17.66,3C17.41,3 17.15,3.1 16.96,3.29L15.13,5.12L18.88,8.87L20.71,7.04C21.1,6.65 21.1,6 20.71,5.63L18.37,3.29C18.17,3.09 17.92,3 17.66,3M14.06,6.19L3,17.25V21H6.75L17.81,9.94L14.06,6.19Z"
                fill="currentColor"
              />
            </svg>
          </a>
        </div>
      </div>
      <div
        className="user-nav"
      >
        <a
          className="user-points"
          href="/@foo/points"
          onClick={[Function]}
          title="You have unclaimed points"
        >
          <span
            className="reward-badge"
          />
          <svg
            viewBox="0 0 24 24"
          >
            <path
              d="M20,14H4V8H9.08L7,10.83L8.62,12L11,8.76L12,7.4L13,8.76L15.38,12L17,10.83L14.92,8H20M20,19H4V17H20M9,4A1,1 0 0,1 10,5A1,1 0 0,1 9,6A1,1 0 0,1 8,5A1,1 0 0,1 9,4M15,4A1,1 0 0,1 16,5A1,1 0 0,1 15,6A1,1 0 0,1 14,5A1,1 0 0,1 15,4M20,6H17.82C17.93,5.69 18,5.35 18,5A3,3 0 0,0 15,2C13.95,2 13.04,2.54 12.5,3.35L12,4L11.5,3.34C10.96,2.54 10.05,2 9,2A3,3 0 0,0 6,5C6,5.35 6.07,5.69 6.18,6H4C2.89,6 2,6.89 2,8V19C2,20.11 2.89,21 4,21H20C21.11,21 22,20.11 22,19V8C22,6.89 21.11,6 20,6Z"
              fill="currentColor"
            />
          </svg>
        </a>
        <a
          className="user-wallet"
          href="/@foo/wallet"
          onClick={[Function]}
          title="Wallet"
        >
          <svg
            viewBox="0 0 24 24"
          >
            <path
              d="M20,8H4V6H20M20,18H4V12H20M20,4H4C2.89,4 2,4.89 2,6V18A2,2 0 0,0 4,20H20A2,2 0 0,0 22,18V6C22,4.89 21.1,4 20,4Z"
              fill="currentColor"
            />
          </svg>
        </a>
        <span
          className="notifications"
          onClick={[Function]}
          title="Notifications"
        >
          <svg
            viewBox="0 0 24 24"
          >
            <path
              d="M21,19V20H3V19L5,17V11C5,7.9 7.03,5.17 10,4.29C10,4.19 10,4.1 10,4A2,2 0 0,1 12,2A2,2 0 0,1 14,4C14,4.1 14,4.19 14,4.29C16.97,5.17 19,7.9 19,11V17L21,19M14,21A2,2 0 0,1 12,23A2,2 0 0,1 10,21"
              fill="currentColor"
            />
          </svg>
        </span>
        <div
          className="custom-dropdown float-right"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <span
            className="user-avatar medium"
            style={
              Object {
                "backgroundImage": "url(https://images.ecency.com/u/foo/avatar/medium)",
              }
            }
          />
        </div>
      </div>
    </div>
    <audio
      autoPlay={false}
      id="notification-audio"
      muted={true}
      src="notification.mp3"
      style={
        Object {
          "display": "none",
        }
      }
    />
  </div>,
]
`;

exports[`(4) With active user && usePrivate = false 1`] = `
Array [
  <div
    className="nav-bar-toggle"
    onClick={[Function]}
  >
    <svg
      viewBox="0 0 24 24"
    >
      <path
        d="M3,6H21V8H3V6M3,11H21V13H3V11M3,16H21V18H3V16Z"
        fill="currentColor"
      />
    </svg>
  </div>,
  <div
    className="nav-bar-sm"
  >
    <div
      className="brand"
    >
      <a
        href="/@foo/feed"
        onClick={[Function]}
      >
        <img
          alt="Logo"
          className="logo"
          src="logo-circle.svg"
        />
      </a>
    </div>
    <div
      className="text-menu"
    >
      <a
        className="menu-item"
        href="/discover"
        onClick={[Function]}
      >
        Discover
      </a>
      <a
        className="menu-item"
        href="/communities"
        onClick={[Function]}
      >
        Communities
      </a>
    </div>
  </div>,
  <div
    className="nav-bar"
  >
    <div
      className="nav-bar-inner"
    >
      <div
        className="brand"
      >
        <a
          href="/@foo/feed"
          onClick={[Function]}
        >
          <img
            alt="Logo"
            className="logo"
            src="logo-circle.svg"
          />
        </a>
      </div>
      <div
        className="text-menu"
      >
        <a
          className="menu-item"
          href="/discover"
          onClick={[Function]}
        >
          Discover
        </a>
        <a
          className="menu-item"
          href="/communities"
          onClick={[Function]}
        >
          Communities
        </a>
      </div>
      <div
        className="flex-spacer"
      />
      <div
        className="search-bar"
      >
        <div
          className="suggestion"
        >
          <div
            className="search-box"
          >
            <span
              className="prepend"
            >
              <svg
                viewBox="0 0 24 24"
              >
                <path
                  d="M9.5,3A6.5,6.5 0 0,1 16,9.5C16,11.11 15.41,12.59 14.44,13.73L14.71,14H15.5L20.5,19L19,20.5L14,15.5V14.71L13.73,14.44C12.59,15.41 11.11,16 9.5,16A6.5,6.5 0 0,1 3,9.5A6.5,6.5 0 0,1 9.5,3M9.5,5C7,5 5,7 5,9.5C5,12 7,14 9.5,14C12,14 14,12 14,9.5C14,7 12,5 9.5,5Z"
                  fill="currentColor"
                />
              </svg>
            </span>
            <input
              autoComplete="off"
              className="form-control"
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Search 10,000,000 posts"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        className="switch-menu"
      >
        <div
          className="switch-language"
        >
          <div
            className="custom-dropdown float-right"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="dropdown-btn"
            >
              
              <div
                className="menu-down"
              >
                <svg
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12.87,15.07L10.33,12.56L10.36,12.53C12.1,10.59 13.34,8.36 14.07,6H17V4H10V2H8V4H1V6H12.17C11.5,7.92 10.44,9.75 9,11.35C8.07,10.32 7.3,9.19 6.69,8H4.69C5.42,9.63 6.42,11.17 7.67,12.56L2.58,17.58L4,19L9,14L12.11,17.11L12.87,15.07M18.5,10H16.5L12,22H14L15.12,19H19.87L21,22H23L18.5,10M15.88,17L17.5,12.67L19.12,17H15.88Z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          className="switch-theme"
          onClick={[Function]}
          title="Night Theme"
        >
          <svg
            viewBox="0 0 24 24"
          >
            <path
              d="M12,18V6A6,6 0 0,1 18,12A6,6 0 0,1 12,18M20,15.31L23.31,12L20,8.69V4H15.31L12,0.69L8.69,4H4V8.69L0.69,12L4,15.31V20H8.69L12,23.31L15.31,20H20V15.31Z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
      <div
        className="btn-menu"
      >
        <div
          className="submit-post"
        >
          <a
            className="btn btn-outline-primary"
            href="/submit"
            onClick={[Function]}
            title="Create Post"
          >
            <svg
              viewBox="0 0 24 24"
            >
              <path
                d="M14.06,9L15,9.94L5.92,19H5V18.08L14.06,9M17.66,3C17.41,3 17.15,3.1 16.96,3.29L15.13,5.12L18.88,8.87L20.71,7.04C21.1,6.65 21.1,6 20.71,5.63L18.37,3.29C18.17,3.09 17.92,3 17.66,3M14.06,6.19L3,17.25V21H6.75L17.81,9.94L14.06,6.19Z"
                fill="currentColor"
              />
            </svg>
          </a>
        </div>
      </div>
      <div
        className="user-nav"
      >
        <a
          className="user-wallet"
          href="/@foo/wallet"
          onClick={[Function]}
          title="Wallet"
        >
          <svg
            viewBox="0 0 24 24"
          >
            <path
              d="M20,8H4V6H20M20,18H4V12H20M20,4H4C2.89,4 2,4.89 2,6V18A2,2 0 0,0 4,20H20A2,2 0 0,0 22,18V6C22,4.89 21.1,4 20,4Z"
              fill="currentColor"
            />
          </svg>
        </a>
        <div
          className="custom-dropdown float-right"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <span
            className="user-avatar medium"
            style={
              Object {
                "backgroundImage": "url(https://images.ecency.com/u/foo/avatar/medium)",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>,
]
`;
